lr-cli/cortx_setup/commands/storage/config.py

```diff
@@ -259,22 +345,53 @@ def run(self, **kwargs):
                     if self.enclosure_id:
                         self.update_pillar_and_conf('name', name)
                     else:
-                        self.logger.error("Please set 'user, password, primary ip and port' first")
-                        raise RuntimeError("Cannot set name before setting user, password, ip and port")
+                        self.logger.error("Enclosure ID is not set"
+                            "Run following command to set the enclosure id:"
+                            "cortx_setup storage config --user <user>"
+                            " --password <passwd> --ip <ip> --port <port>")
+                        raise RuntimeError(
+                            "Cannot set enclosure name without enclosure id"
+                        )
 
                 if storage_type is not None:
                     if self.enclosure_id:
```

Comment:
Do we not have default enclosure_id for VM?

Reply:
The enclosure_id in case of VM is:  **enc_+machine_id** which is set at the time of storage config.
```
cortx_setup storage config --name <name> --type <type> --controller_type <controller_type>
```